src/iceberg/manifest_list.cc

```diff
@@ -0,0 +1,105 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include "iceberg/manifest_list.h"
+
+#include <vector>
+
+#include "iceberg/schema_field.h"
+#include "iceberg/type.h"
+
+namespace iceberg {
+
+const SchemaField FieldSummary::CONTAINS_NULL =
+    SchemaField::MakeRequired(509, "contains_null", std::make_shared<BooleanType>());
+const SchemaField FieldSummary::CONTAINS_NAN =
+    SchemaField::MakeOptional(518, "contains_nan", std::make_shared<BooleanType>());
+const SchemaField FieldSummary::LOWER_BOUND =
+    SchemaField::MakeOptional(510, "lower_bound", std::make_shared<BinaryType>());
+const SchemaField FieldSummary::UPPER_BOUND =
+    SchemaField::MakeOptional(511, "upper_bound", std::make_shared<BinaryType>());
+
+StructType FieldSummary::GetType() {
+  return StructType({
+      CONTAINS_NULL,
+      CONTAINS_NAN,
+      LOWER_BOUND,
+      UPPER_BOUND,
+  });
+}
+
+const SchemaField ManifestFile::MANIFEST_PATH =
```

Comment:
I'm not sure if using macros is a good idea in modern C++ to replace these repeated lengthy lines. 

@lidavidm WDYT?